src/transformers/models/bert/modeling_bert.py

```diff
@@ -240,6 +240,22 @@ def forward(
         return embeddings
 
 
+class QATMatMul(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+        # behaves like normal torch.matmul unless a SparseML QuantizationModifier
+        # is initialized
+        self.wrap_qat = True
+        self.qat_wrapper_kwargs = {
+            "num_inputs": 2,
+            "input_qconfigs": ["asymmetric", "symmetric"],
+        }
+
+    def forward(self, a: torch.Tensor, b: torch.Tensor):
+        return torch.matmul(a, b)
+
+
 class BertSelfAttention(nn.Module):
```

Comment:
Yeah I'm almost positive we could hot patch this stuff in instead of having repo forks.

Something like

```
import transformers

from transformers.models.bert.modeling_bert import BertSelfAttention as _BertSelfAttention

class PatchedBertSelfAttention(_BertSelfAttention):
    def __init__(self, *args, **kwargs):
        self.attention_scores_matmul = ...
    def forward(self):
        ...

transformers.models.bert.modeling_bert.BertSelfAttention = PatchedBertSelfAttention
```

Reply:
Yeah I think now that most of the implementation has been moved on the sparseml side, there's definitely potential to explore here. One thing to keep in mind is that every time we've upgraded the HF repo had changes that broke out integration, so the hot patch would need to be easily to debug and amend. But in general I'm all for this